arch/risc-v/src/esp32c3/esp32c3_rtc_lowerhalf.c

```diff
@@ -0,0 +1,575 @@
+static int rtc_setalarm(FAR struct rtc_lowerhalf_s *lower,
+                        FAR const struct lower_setalarm_s *alarminfo);
+static int rtc_setrelative(FAR struct rtc_lowerhalf_s *lower,
+                           FAR const struct lower_setrelative_s *alarminfo);
+static int rtc_cancelalarm(FAR struct rtc_lowerhalf_s *lower,
+                           int alarmid);
+static int rtc_rdalarm(FAR struct rtc_lowerhalf_s *lower,
+                       FAR struct lower_rdalarm_s *alarminfo);
+#endif
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+/* ESP32-C3 RTC driver operations */
+
+static const struct rtc_ops_s g_rtc_ops =
+{
+  .rdtime      = rtc_rdtime,
+  .settime     = rtc_settime,
+  .havesettime = rtc_havesettime,
+#ifdef CONFIG_RTC_ALARM
+  .setalarm    = rtc_setalarm,
+  .setrelative = rtc_setrelative,
+  .cancelalarm = rtc_cancelalarm,
+  .rdalarm     = rtc_rdalarm,
+#endif
+#ifdef CONFIG_RTC_PERIODIC
+  .setperiodic    = NULL,
+  .cancelperiodic = NULL,
+#endif
+#ifdef CONFIG_RTC_IOCTL
+  .ioctl       = NULL,
+#endif
+#ifndef CONFIG_DISABLE_PSEUDOFS_OPERATIONS
+  .destroy     = NULL,
+#endif
+};
+
+/* ESP32-C3 RTC device state */
+
+static struct esp32c3_lowerhalf_s g_rtc_lowerhalf =
+{
+  .ops        = &g_rtc_ops,
+};
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: rtc_alarm_callback
+ *
+ * Description:
+ *   This is the function that is called from the RTC driver when the alarm
+ *   goes off. It just invokes the upper half drivers callback.
+ *
+ * Returned Value:
+ *   None
+ *
+ ****************************************************************************/
+
+#ifdef CONFIG_RTC_ALARM
+static void rtc_alarm_callback(FAR void *arg, unsigned int alarmid)
+{
+  FAR struct esp32c3_lowerhalf_s *lower;
+  FAR struct esp32c3_cbinfo_s *cbinfo;
+  rtc_alarm_callback_t cb;
+  FAR void *priv;
+
+  DEBUGASSERT((RTC_ALARM0 <= alarmid) && (alarmid < RTC_ALARM_LAST));
+
+  lower        = (struct esp32c3_lowerhalf_s *)arg;
+  cbinfo       = &lower->cbinfo[alarmid];
+
+  /* Sample and clear the callback information to minimize the window in
+   * time in which race conditions can occur.
+   */
+
+  cb           = (rtc_alarm_callback_t)cbinfo->cb;
+  priv         = (FAR void *)cbinfo->priv;
+
+  cbinfo->cb   = NULL;
+  cbinfo->priv = NULL;
+
+  /* Perform the callback */
+
+  if (cb != NULL)
+    {
+      cb(priv, alarmid);
+    }
+}
+#endif /* CONFIG_RTC_ALARM */
+
+/****************************************************************************
+ * Name: rtc_rdtime
+ *
+ * Description:
+ *   Implements the rdtime() method of the RTC driver interface
```

Comment:
I suggest to use meaninfull descriptions here.
I often take them from the driver peripheral header file in the section "IOCTLS":
PTAL: https://github.com/apache/incubator-nuttx/blob/master/include/nuttx/timers/rtc.h#L109
For example:
```suggestion
 *   Returns the current RTC time.
```

Reply:
Done